### PR TITLE
Remove unnecessary keys and values from simulator output

### PR DESCRIPTION
**Pull request by nagkumar91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py [hunk 1]
```diff
@@ -193,6 +193,12 @@ def _to_chat_protocol(self, *, conversation_history, template_parameters):
             del template_parameters["topic"]
         if "ch_template_placeholder" in template_parameters:
             del template_parameters["ch_template_placeholder"]
+        if "conversation_starter" in template_parameters:
```
Comment (nick863) [inline src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py:196]: Nit: May be we should do 
```python
for key in ('conversation_starter', 'group_of_people', 'metadata'):
    template_parameters.pop(key, None)
```

Comment (nagkumar91) [reply src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py]: changed :)

File: src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py [hunk 2]
```diff
@@ -187,12 +187,16 @@ def _to_chat_protocol(self, *, conversation_history, template_parameters):
...
-            del template_parameters["ch_template_placeholder"]
+        for key in (
+            "conversation_starter",
+            "group_of_people",
+            "metadata",
+            "target_population",
+            "topic",
+            "ch_template_placeholder",
+        ):
+            if key in template_parameters:
```
Comment (nick863) [inline src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py:198]: Nit: if using pop, you will not need the check for value:
template_parameters.pop(key, None)

**Review by nick863 — CHANGES_REQUESTED** [review]

File: src/promptflow-evals/promptflow/evals/evaluators/_content_safety/_content_safety.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 
 class ContentSafetyEvaluator:
-    def __init__(self, project_scope: dict, parallel: bool = True, credential=None):
```
Comment (ninghu) [inline src/promptflow-evals/promptflow/evals/evaluators/_content_safety/_content_safety.py:10]: why this change?

File: src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py (hunk 2 above)
Comment (nagkumar91) [reply src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py:198]: I didn't know that! fixing it

Comment (nick863) [reply src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py:198]: In this case if key is not there, the default (None) will be returned.

File: src/promptflow-evals/promptflow/evals/evaluators/_content_safety/_content_safety.py (hunk 3 above)
Comment (nagkumar91) [reply src/promptflow-evals/promptflow/evals/evaluators/_content_safety/_content_safety.py]: Pushed this by mistake. Reverting it.

File: src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py (hunk 2 above)
Comment (nagkumar91) [reply src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py]: Done 👍🏽

**Review by nick863 — APPROVED** [review]

**Review by ninghu — APPROVED** [review]

**Review by trangevi — APPROVED** [review]
